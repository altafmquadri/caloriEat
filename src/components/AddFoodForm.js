import React, { Component } from 'react';
import { connect } from 'react-redux'
import moment from 'moment'
import 'react-dates/initialize'
import 'react-dates/lib/css/_datepicker.css'
import { SingleDatePicker } from 'react-dates'
import { addFood } from './actions/food'
import { addMeal } from './actions/meal'

class AddFoodForm extends Component {
    state = {
        date: moment(),
        mealCategory: 'breakfast',
        calendarFocused: false,
    }

    onDateChange = date => {
        if (date) this.setState({ date })
    }

    onSubmit = e => {
        e.preventDefault()
        const food = {
            date: this.state.date,
            mealCategory: this.state.mealCategory
        }
        this.props.addFood(food) //from mapDispatch to props updates the food state, viewing in redux tools shows proper state for food
        const meal = this.props.food //comes from mapStateToProps 
        console.log(this.props, 'logging props before meal assignment');
        console.log(meal, 'in form') //logging this shows it's not the updated state I am expecting
        this.props.addMeal(meal) //trying to add state from food reducer to the meal reducer, meal reducer is not updated via redux tools
    }

    onFocusChange = ({ focused }) => this.setState({ calendarFocused: focused })
    onMealChange = e => this.setState({ mealCategory: e.target.value })

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <select value={this.state.mealCategory}
                        onChange={this.onMealChange}>
                        <option value='breakfast'>Breakfast</option>
                        <option value='lunch'>Lunch</option>
                        <option value='dinner'>Dinner</option>
                        <option value='snack'>Snack</option>
                    </select>
                    <SingleDatePicker
                        date={this.state.date}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        id="caloriEat-addFood-form" />
                    <button type="submit">Add Food</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addFood: food => dispatch(addFood(food)),
    addMeal: meal => dispatch(addMeal(meal))
})

const mapStateToProps = state => ({
    food: state.food
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodForm)