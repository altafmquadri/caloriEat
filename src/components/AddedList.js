import React from 'react';
import { connect } from 'react-redux'

export const AddedList = (props) => {
    return (
        <div>
            {props.list.map(meal => (
                <li key={meal.uuid}>{`${meal.foodItem} added, calories:${meal.calories}`}</li>
            ))}
        </div>
    );
}

const mapStateToProps = (state) => ({
    list: state.list
})

export default connect(mapStateToProps)(AddedList)