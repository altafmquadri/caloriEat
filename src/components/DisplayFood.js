import React, { Component } from 'react';
import { connect } from 'react-redux'
import { fetchServing } from './actions/food'


class DisplayFood extends Component {
    state = {
        quantity: '',
        measureURI: '',
        foodId: ''
    }

    onChange = e => {
        this.setState({
            quantity: 1,
            measureURI: e.target.value,
            foodId: this.props.food.foodId
        }, () => this.props.fetchServing(this.state))
    }

    render() {
        console.log(this.props);
        const displayContents = () => {
            if (this.props.food.foodItem) {
                const { foodItem, calories, carbohydrates, fat, protein, fiber, image, measures } = this.props.food
                const contents = (
                    <div>
                        <h1>{foodItem}</h1>
                        <h3>Calories: {calories}</h3>
                        <h3>Carbohydrates: {carbohydrates}g</h3>
                        <h3>Fat: {fat}g</h3>
                        <h3>Protein: {protein}g</h3>
                        <h3>Fiber: {fiber}g</h3>
                        <img alt={foodItem} src={image}></img>
                        <select
                            onChange={this.onChange}
                            value={this.state.measureURI}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                    </div >
                )
                return contents
            } else return null

        }
        console.log(this.props);
        return (
            <div>
                {displayContents()}
            </div>

        )
    }
}




const mapStateToProps = (state) => ({
    food: state.food
})

const mapPropsToDispatch = (dispatch) => ({
    fetchServing: obj => dispatch(fetchServing(obj))
})

export default connect(mapStateToProps, mapPropsToDispatch)(DisplayFood)