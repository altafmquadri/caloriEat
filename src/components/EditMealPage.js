import React, { Component } from 'react';
import { connect } from 'react-redux'
import { editServing } from './actions/meal';

class EditMealPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            uuid: this.props.meal.uuid,
            foodId: this.props.meal.foodId,
            measureURI: this.props.meal.measureURI,
            quantity: this.props.meal.quantity,
            mealCategory: this.props.meal.mealCategory
        }
    }

    onMeasureChange = e => (this.setState({ measureURI: e.target.value }))
    onQuantityChange = e => (this.setState({ quantity: e.target.value }))
    onMealCategoryChange = e => (this.setState({ mealCategory: e.target.value }))

    onSubmit = e => {
        e.preventDefault()
        const { foodId, measureURI, quantity, uuid, mealCategory } = this.state
        const editsObj = {
            foodId,
            measureURI,
            quantity: parseInt(quantity)
        }
        this.props.editServing(uuid, editsObj, mealCategory)

    }

    render() {
        const { foodItem, calories, measures } = this.props.meal
        const { measureURI, quantity, mealCategory } = this.state
        return (
            <div>
                <h1>Edit Meal</h1>
                <h3>{foodItem}</h3>
                <h3>{calories}</h3>
                <div>
                    <form onSubmit={this.onSubmit}>
                        <select
                            value={measureURI}
                            onChange={this.onMeasureChange}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                        <input
                            value={quantity}
                            onChange={this.onQuantityChange}
                            type="text"></input>
                        <select value={mealCategory}
                            onChange={this.onMealCategoryChange}>
                            <option value='breakfast'>Breakfast</option>
                            <option value='lunch'>Lunch</option>
                            <option value='dinner'>Dinner</option>
                            <option value='snack'>Snack</option>
                        </select>
                        <button type="submit">Edit Meal</button>
                    </form>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    meal: state.meals.find(meal => meal.uuid === props.match.params.uuid)
})

const mapDispatchToProps = dispatch => ({
    editServing: (uuid, fetchObj, mealCategory) => dispatch(editServing(uuid, fetchObj, mealCategory))
})
export default connect(mapStateToProps, mapDispatchToProps)(EditMealPage)