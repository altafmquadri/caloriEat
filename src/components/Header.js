import React from 'react';
import { NavLink } from 'react-router-dom'

const Header = () => {
    return (
        <div>
            <NavLink to='/'>Home</NavLink>
            <NavLink to='/search'>Search</NavLink>
        </div>
    );
}

export default Header;