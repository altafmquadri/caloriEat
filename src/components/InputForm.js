import React, { Component } from 'react';
import { connect } from 'react-redux'
import { fetchFood } from './actions/food'

class InputForm extends Component {
    state = {
        foodItem: ''
    }

    onChange = e => {
        this.setState({ foodItem: e.target.value });
    }

    onSubmit = e => {
        e.preventDefault()
        const lookup = this.state.foodItem.split(' ').join('%20')
        this.props.fetchFood(lookup)
    }

    render() {
        return (
            <form onSubmit={this.onSubmit}>
                <input
                    type="text"
                    placeholder="enter food item here"
                    value={this.state.foodItem}
                    onChange={this.onChange}></input>
                <button type="submit">Find Food</button>
            </form>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchFood: lookup => dispatch(fetchFood(lookup))
    }
}

export default connect(undefined, mapDispatchToProps)(InputForm)