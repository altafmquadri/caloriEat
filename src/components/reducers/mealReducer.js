const mealReducer = (state = [], action) => {
    switch (action.type) {
        case 'ADD_MEAL': {
            const meal = action.payload
            return [...state, meal]
        }
        case 'EDIT_MEAL': {
            const {
                calories,
                totalNutrients: {
                    CHOCDF: { quantity: carbohydrates },
                    FAT: { quantity: fat },
                    FIBTG: { quantity: fiber },
                    PROCNT: { quantity: protein },
                }
            } = action.payload
            const item = {
                ...state.find(meal => meal.uuid === action.uuid),
                ...action.properties,
                calories,
                carbohydrates: parseFloat(carbohydrates).toFixed(2),
                fat: parseFloat(fat).toFixed(2),
                fiber: parseFloat(fiber).toFixed(2),
                protein: parseFloat(protein).toFixed(2),
            }
            return state.map(meal => {
                if (meal.uuid === action.uuid) {
                    return meal = { ...meal, ...item }
                }
                return meal
            })
        }
        case 'DELETE_MEAL': {
            const id = action.payload
            return state.filter(meal => meal.uuid !== id)
        }
        default:
            return state
    }
}

export default mealReducer