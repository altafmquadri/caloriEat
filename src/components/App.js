import React, { Component } from 'react';
import InputForm from './InputForm'
import DisplayFood from './DisplayFood'

class App extends Component {
    render() {
        return (
            <div>
                <InputForm />
                <DisplayFood />
            </div>
        )
    }
}

export default App;