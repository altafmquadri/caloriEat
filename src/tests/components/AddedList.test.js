import React from 'react';
import { shallow } from 'enzyme'
import { AddedList } from '../../components/AddedList'
import {meals} from '../fixtures/mealReducer'

describe('<AddedList/>', () => {
    test('renders AddedList when there are no meals added', () => {
        const wrapper = shallow(<AddedList list={[]} />)
        expect(wrapper).toMatchSnapshot()
    })
    test('renders AddedList when there are meals added', () => {
        const wrapper = shallow(<AddedList list={meals} />)
        expect(wrapper).toMatchSnapshot()
    })
})