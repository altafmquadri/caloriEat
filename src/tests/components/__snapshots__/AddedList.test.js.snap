// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddedList/> renders AddedList when there are meals added 1`] = `
<div>
  <li
    key="1"
  >
    milk added, calories:148
  </li>
  <li
    key="2"
  >
    oreo added, calories:55
  </li>
  <li
    key="3"
  >
    french fries added, calories:255
  </li>
</div>
`;

exports[`<AddedList/> renders AddedList when there are no meals added 1`] = `<div />`;
